packages/cli-kit/src/public/node/api/partners.ts

```diff
@@ -38,42 +38,37 @@ export async function partnersRequest<T>(query: string, token: string, variables
   return result
 }
 
-interface ProxyResponse {
-  scriptServiceProxy: string
+interface FunctionUploadUrlGenerateResponse {
+  uploadUrlGenerate: {
+    url: string
+    moduleId: string
+    headers: string
+    maxBytes: number
+    maxSize: string
+  }
 }
 
 /**
- * Function queries are proxied through the script service proxy.
- * To execute a query, we encapsulate it inside another query (including the variables)
- * This is done automatically, you just need to provide the query and the variables.
+ * Request a URL from partners to which we will upload our function.
  *
- * @param apiKey - APIKey of the app where the query will be executed.
- * @param query - GraphQL query to execute.
  * @param token - Partners token.
- * @param variables - GraphQL variables to pass to the query.
  * @returns The response of the query.
  */
-export async function functionProxyRequest<T>(
-  apiKey: string,
-  query: unknown,
-  token: string,
-  variables?: unknown,
-): Promise<T> {
-  const proxyVariables = {
-    api_key: apiKey,
-    query,
-    variables: JSON.stringify(variables) || '{}',
-  }
-  const proxyQuery = ScriptServiceProxyQuery
-  const res: ProxyResponse = await partnersRequest(proxyQuery, token, proxyVariables)
-  const json = JSON.parse(res.scriptServiceProxy)
-  handleDeprecations(json)
-  return json as T
+export async function getFunctionUploadUrl<T>(token: string): Promise<T> {
```

Comment:
Follow-up to [this comment](https://github.com/Shopify/cli/pull/3228/files#r1450554945), I think the sig will look like this:

```suggestion
export async function getFunctionUploadUrl(token: string): Promise<FunctionUploadUrlGenerateResponse> {
```